src/reducers/collapseContext.ts

```diff
@@ -1,32 +1,50 @@
-import { parentOf, pathToContext, reducerFlow, unroot } from '../util'
+import { head, parentOf, pathToContext, reducerFlow, unroot } from '../util'
 import { State } from '../util/initialState'
-import { getChildren } from '../selectors'
+import { getAllChildren, simplifyPath } from '../selectors'
 import { archiveThought, existingThoughtMove, setCursor } from '../reducers'
+import _ from 'lodash'
+import existingThoughtDelete from './existingThoughtDelete'
+import { Path } from '../types'
+
+interface Options {
+  deleteCursor?: boolean,
+  at?: Path | null,
+}
 
 /** Collapses the active thought. */
-const collapseContext = (state: State) => {
+const collapseContext = (state: State, { deleteCursor, at }: Options) => {
   const { cursor } = state
 
-  if (!cursor) return state
+  const path = at || cursor
+
+  if (!path) return state
+
+  const simpleCursor = simplifyPath(state, path)
```

Comment:
Since `path` may not be the `cursor` any more, we should call this `simplePath`.